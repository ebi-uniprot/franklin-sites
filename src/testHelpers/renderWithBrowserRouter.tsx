import { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';

// Inspired by https://testing-library.com/docs/example-react-router/

const renderWithBrowserRouter = (ui: ReactElement, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return render(ui, { wrapper: BrowserRouter });
};

export default renderWithBrowserRouter;
