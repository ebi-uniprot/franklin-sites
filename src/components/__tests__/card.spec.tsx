import Card from '../card';

import renderWithBrowserRouter from '../../testHelpers/renderWithBrowserRouter';

describe('Card component', () => {
  it('should render', () => {
    const { asFragment } = renderWithBrowserRouter(
      <Card header={<h2>Title</h2>}>
        <span>Some content</span>
      </Card>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render card with links', () => {
    const links = [
      {
        name: 'link',
        link: 'example.com',
        color: 'red',
      },
    ];

    const { asFragment } = renderWithBrowserRouter(
      <Card header={<h2>Title</h2>} links={links}>
        <span>Some content</span>
      </Card>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
