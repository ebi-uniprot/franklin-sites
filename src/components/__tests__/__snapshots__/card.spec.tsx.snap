// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render 1`] = `
<DocumentFragment>
  <section
    class="card"
  >
    <div
      class="card__container"
    >
      <div
        class="card__header card__header--with-separator"
      >
        <h2>
          Title
        </h2>
      </div>
      <div
        class="card__content"
      >
        <span>
          Some content
        </span>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Card component should render card with links 1`] = `
<DocumentFragment>
  <section
    class="card"
  >
    <div
      class="card__container"
    >
      <div
        class="card__header card__header--with-separator"
      >
        <h2>
          Title
        </h2>
      </div>
      <div
        class="card__content"
      >
        <span>
          Some content
        </span>
      </div>
    </div>
    <div
      class="card__actions"
    >
      <a
        href="https://www.uniprot.org"
      >
        link
      </a>
    </div>
  </section>
</DocumentFragment>
`;
